src/Codeception/Module/REST.php

```diff
@@ -841,6 +843,57 @@ public function seeResponseContainsJson($json = [])
         );
     }
 
+    /**
+     * Checks whether last response matches the supplied json schema
+     * Supply schema as json string
+     *
+     * @part json
+     */
+    public function seeResponseIsValidOnJsonSchema($schema)
+    {
+        $responseContent = $this->connectionModule->_getResponseContent();
+        \PHPUnit\Framework\Assert::assertNotEquals('', $responseContent, 'response is empty');
+        $responseObject = json_decode($responseContent);
+        $errorCode = json_last_error();
+        $errorMessage = json_last_error_msg();
+        \PHPUnit\Framework\Assert::assertEquals(
+            JSON_ERROR_NONE,
+            $errorCode,
+            sprintf(
+                "Invalid json: %s. System message: %s.",
+                $responseContent,
+                $errorMessage
+            )
+        );
```

Comment:
This block of code is identical copy of `seeResponseIsJson` method,
the block below is the same, but it validates schema instead of respose,
it would be good to extract common parts to `private isValidJson($text)` method.

Reply:
Absolutely not sure why I didn’t do that in the first place

Reply:
Fixed